### PR TITLE
kubeadm: fix kubeadm reset logic

### PR DESCRIPTION
**Pull request by bart0sh**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

If /etc/kubeadm/amdin.conf doesn't exist kubeadm reset fails
with the error:
    failed to load admin kubeconfig: open /root/.kube/config: no such file or directory

Fixed by properly checking if file exists before using it.

**Special notes for your reviewer**:

This issue was caused by [my previous PR](https://github.com/kubernetes/kubernetes/pull/73636), so I reverted it and added check for client variable.

However, I've noticed that cmdutil.FindExistingKubeConfig logic is broken by design as
- it returns file path that doesn't exist, which is opposite to what its name is suggesting
- it returns /root/.kube/config, because kubeadm is usually run with sudo. This function uses k8s.io/client-go/tools/clientcmd APIs, which use Getenv('HOME') as one of default locations. It should look at the current user home instead, I believe. This can be done by using SUDO_USER variable if it's set. I'm not sure where this should be fixed - in cmdutil.FindExistingKubeConfig or in clientcmd API.

Another important thing that is worth mentioning: new resetDetectCRISocket function tries to fetch InitConfiguration from the cluster, which in most cases doesn't exist yet.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by neolit123 — APPROVED** [review]
this change LGTM thanks @bart0sh 

`FindExistingKubeConfig()` is in a weird space ATM and there is one open PR to fix it, yet the name would be still misleading. the client-go tools are not ideal because they return no indication if a file returned by the precedence logic exists or not. so if we really want to make sure a file exists we should check ourself.

the only reason we are using the client-go tools is to comply with other tools that use them in the ecosystem.

/approve
/priority important-longterm

**Review by rosti — APPROVED** [review]
Thanks @bart0sh !
/lgtm

File: cmd/kubeadm/app/cmd/reset.go [hunk 1]
```diff
@@ -62,8 +62,10 @@ func NewCmdReset(in io.Reader, out io.Writer) *cobra.Command {
 			kubeadmutil.CheckErr(err)
 
 			kubeConfigFile = cmdutil.FindExistingKubeConfig(kubeConfigFile)
-			client, err = getClientset(kubeConfigFile, false)
-			kubeadmutil.CheckErr(err)
+			if _, err := os.Stat(kubeConfigFile); !os.IsNotExist(err) {
```
Comment (yagonobre) [inline cmd/kubeadm/app/cmd/reset.go]: Should we test just `err != nil`?

**Review by yagonobre — COMMENTED** [review]
/hold

I guess that `!os.IsNotExist(err)` will try to create the client even if we have a different error like permission issue.

**Comment by bart0sh** [comment]
@yagonobre 

> I guess that `!os.IsNotExist(err)` will try to create the client even if we have a different error like permission issue.

the code tries to create client if configuration file exists. If there are other problems it should be reported by the getClientset code. If we do as you're suggesting we'll effectively ignore all possible issues.

Does this make sense?

File: cmd/kubeadm/app/cmd/reset.go (hunk 1 above)
Comment (rosti) [reply cmd/kubeadm/app/cmd/reset.go]: I don't think it's safe to continue with reset if we encounter something different than ErrNotExist. The rest of the [stat(2)](http://man7.org/linux/man-pages/man2/stat.2.html) possible errors are bad enough for us to stop after the call to `getClientset`.

**Comment by yagonobre** [comment]
@bart0sh Make senses, Sorry for this I just wanted to be sure.

File: cmd/kubeadm/app/cmd/reset.go (hunk 1 above)
Comment (bart0sh) [reply cmd/kubeadm/app/cmd/reset.go]: @rosti 
> The rest of the [stat(2)](http://man7.org/linux/man-pages/man2/stat.2.html) possible errors are bad enough for us to stop after the call to `getClientset`

isn't is what kubeadmutil.CheckErr(err) is called for?

**Pull request merged**